packages/design/src/physics/shape/IColliderShape.ts

```diff
@@ -41,8 +41,31 @@ export interface IColliderShape {
    */
   setIsTrigger(value: boolean): void;
 
+  /**
+   * Get the distance between a point and the shape.
+   * @param position - The position in world space
+   * @param rotation - The rotation in world space
+   * @param point - The point
+   * @returns The distance information
+   */
+  pointDistance(position: Vector3, rotation: Quaternion, point: Vector3): IPointDistanceInfo;
   /**
    * Decrements the reference count of a shape and releases it if the new reference count is zero.
    */
   destroy(): void;
 }
+
+/**
+ * Distance information of a point to the shape.
+ */
+export interface IPointDistanceInfo {
```

Comment:
Can be merged into a Vector4. For internal interface specifications, appropriate trade-offs can be made.